kolibri/core/assets/src/views/k-checkbox/index.vue

```diff
@@ -0,0 +1,203 @@
+          class="k-checkbox-checked"
+        />
+        <mat-svg
+          v-else
+          category="toggle"
+          name="check_box_outline_blank"
+          class="k-checkbox-unchecked"
+        />
+
+      </div>
+
+      <label
+        v-if="label"
+        class="k-checkbox-label"
+        :for="id"
+        :class="{ 'visuallyhidden': !showLabel }"
+      >
+        {{ label }}
+      </label>
+
+    </div>
+  </div>
+
+</template>
+
+
+<script>
+
+  /**
+    * Used for selection
+    */
+  export default {
+    name: 'k-checkbox',
+    props: {
+      /**
+       * Label
+       */
+      label: {
+        type: String,
+        required: true,
+      },
+      /**
+       * Whether to show label
+       */
+      showLabel: {
+        type: Boolean,
+        default: true,
+      },
+      /**
+       * Checked state
+       */
+      checked: {
+        type: Boolean,
+        default: false,
+      },
+      /**
+       * Indeterminate state, overrides checked state
+       */
+      indeterminate: {
+        type: Boolean,
+        default: false,
+      },
+      /**
+       * Disabled state
+       */
+      disabled: {
+        type: Boolean,
+        default: false,
+      },
+    },
+    data: () => ({
+      isCurrentlyChecked: false,
+      isCurrentlyIndeterminate: false,
+      isActive: false,
+    }),
+    computed: {
+      id() {
+        return `k-checkbox-${this._uid}`;
+      },
+    },
+    watch: {
+      checked(newCheckedState) {
+        this.isCurrentlyChecked = newCheckedState;
+      },
+      indeterminate(newIndeterminateState) {
+        this.isCurrentlyIndeterminate = newIndeterminateState;
+      },
+    },
+    created() {
+      this.isCurrentlyChecked = this.checked;
+      this.isCurrentlyIndeterminate = this.indeterminate;
+    },
+    methods: {
+      toggleCheck(event) {
+        if (!this.disabled) {
+          this.isCurrentlyChecked = !this.isCurrentlyChecked;
+          this.isCurrentlyIndeterminate = false;
+          /**
+           * Emits change event
+           */
+          this.$emit('change', this.isCurrentlyChecked, event);
+        }
+      },
+    },
+  };
+
+</script>
+
+
+<style lang="stylus" scoped>
+
+  @require '~kolibri.styles.definitions'
+
+  $checkbox-height = 24px
+
+  .k-checkbox-container
+    display: table
+    margin-top: 8px
+    margin-bottom: 8px
+
+  .tr
+    display: table-row
+
+  .k-checkbox
+    display: table-cell
+    position: relative
+    vertical-align: top
+    width: $checkbox-height
+    height: $checkbox-height
+    cursor: pointer
+
+  .k-checkbox-input
+    position: absolute
+    top: 50%
+    left: 50%
+    transform: translate(-50%, -50%)
+    opacity: 0
+    cursor: pointer
+
+  .k-checkbox-checked, .k-checkbox-indeterminate
+    fill: $core-action-normal
+
+  .k-checkbox-unchecked
+    fill: $core-text-annotation
+
+  .k-checkbox-active
+    .k-checkbox-checked, .k-checkbox-indeterminate
+      fill: $core-action-dark
```

Comment:
I think it makes more sense to replace the 'Select all' with 'Name' since we don't follow that pattern in other places like the enroll and groups page.